pkg/apis/pipeline/v1beta1/pipelinerun_conversion_test.go

```diff
@@ -285,3 +409,263 @@ func TestPipelineRunConversionFromDeprecated(t *testing.T) {
+					},
+				},
+			},
+		}}
+	for _, test := range tests {
+		versions := []apis.Convertible{&v1.PipelineRun{}}
+		for _, version := range versions {
+			t.Run(test.name, func(t *testing.T) {
+				ctx := withMinialEmbeddedStatus(context.Background())
+				ver := version
+
+				if err := test.in.ConvertTo(ctx, ver); err != nil {
+					t.Errorf("ConvertTo() = %v", err)
+				}
+				t.Logf("ConvertTo() = %#v", ver)
+				got := &v1beta1.PipelineRun{}
+
+				if err := got.ConvertFrom(ctx, ver); err != nil {
+					t.Errorf("ConvertFrom() = %v", err)
+				}
+				t.Logf("ConvertFrom() = %#v", got)
+				if d := cmp.Diff(test.want, got); d != "" {
+					t.Errorf("roundtrip %s", diff.PrintWantGot(d))
+				}
+			})
+		}
+	}
+}
+
+func TestPipelineRunConversionEmbeddedStatusError(t *testing.T) {
+	tests := []struct {
+		name           string
+		in             *v1.PipelineRun
+		embeddedStatus string
+		expectedError  error
+	}{{
+		name: "v1 to v1beta1 with full embedded status invalid",
+		in: &v1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "foo",
+				Namespace: "bar",
+			},
+			Spec: v1.PipelineRunSpec{
+				PipelineRef: &v1.PipelineRef{
+					Name: "test",
+				},
+			},
+			Status: v1.PipelineRunStatus{
+				PipelineRunStatusFields: v1.PipelineRunStatusFields{
+					ChildReferences: []v1.ChildStatusReference{{
+						TypeMeta:         runtime.TypeMeta{Kind: "TaskRun", APIVersion: "v1beta1"},
+						Name:             "tr-0",
+						PipelineTaskName: "ptn",
+						WhenExpressions:  []v1.WhenExpression{{Input: "default-value", Operator: "in", Values: []string{"val"}}},
+					}},
+				},
+			},
+		},
+		embeddedStatus: config.FullEmbeddedStatus,
+		expectedError:  fmt.Errorf("PipelineRun .status.taskRuns and .status.runs fields do not exist in v1, and cannot be populated by conversion from a v1 PipelineRun to a v1beta1 PipelineRun when the 'embedded-status' feature flag is not set to 'minimal'"),
+	}, {
+		name: "v1 to v1beta1 status with both embedded status invalid",
+		in: &v1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "foo",
+				Namespace: "bar",
+			},
+			Spec: v1.PipelineRunSpec{
+				PipelineRef: &v1.PipelineRef{
+					Name: "test",
+				},
+			},
+			Status: v1.PipelineRunStatus{
+				PipelineRunStatusFields: v1.PipelineRunStatusFields{
+					ChildReferences: []v1.ChildStatusReference{{
+						TypeMeta:         runtime.TypeMeta{Kind: "TaskRun", APIVersion: "v1beta1"},
+						Name:             "tr-0",
+						PipelineTaskName: "ptn",
+						WhenExpressions:  []v1.WhenExpression{{Input: "default-value", Operator: "in", Values: []string{"val"}}},
+					}},
+				},
+			},
+		},
+		embeddedStatus: config.BothEmbeddedStatus,
+		expectedError:  fmt.Errorf("PipelineRun .status.taskRuns and .status.runs fields do not exist in v1, and cannot be populated by conversion from a v1 PipelineRun to a v1beta1 PipelineRun when the 'embedded-status' feature flag is not set to 'minimal'"),
```

Comment:
nit: I prefer asserting only that there is an error, and maybe a regex match on part of the error string, but not the entire error string itself. The reason is that it makes tests brittle: they will fail if we change only the error string and no other behavior, which doesn't provide much useful signal. However this is subjective/up to you

Reply:
Thanks Lee for the suggestion. Tried to refactor into this way but couldn't find a single phrase that's best suited for the regex matching piece here. And it seems the reason to have such long error message here is both sentences in the message is required:/ Would love to hear if any piece is preferred to be in the regex or I can keep an eye out for the refactorization on this during the pipelineRunStatus migration.

Reply:
I think it would be OK to just have `wantErr: bool`

Reply:
Thanks Lee, updated!